common/map_test.cpp

```diff
@@ -226,6 +226,92 @@ TYPED_TEST(MapTest, Conversions) {
   EXPECT_EQ(104, *cmv3[4]);
 }
 
+TYPED_TEST(MapTest, Grow) {
+  using MapT = TypeParam;
+
+  MapT m;
+  // Grow when empty. May be a no-op for some small sizes.
+  m.Grow(32);
```

Comment:
Is there a way to see the allocated size at this point? Even a test-only API would help.

Reply:
I can now that there are a more useful set of metrics. I was avoiding coupling the two as I wasn't sure it was worthwhile.

Would using the metrics for this kind of thing be reasonable for you?

Reply:
Playing with it, this looks quite good. Switched to checking the growth pretty precisely. PTAL.

Reply:
Great! Could you do the same for `set_test.cpp` please?

Reply:
Doh, could have sworn I did that. Sorry! =D Done.